pyiceberg/table/__init__.py

```diff
@@ -564,41 +584,84 @@ class AssertRefSnapshotId(TableRequirement):
     ref: str
     snapshot_id: Optional[int] = Field(default=None, alias="snapshot-id")
 
+    def validate(self, base_metadata: TableMetadata) -> None:
+        snapshot_ref = base_metadata.refs.get(self.ref)
+        if snapshot_ref is not None:
```

Comment:
Calling in the Walrus operator:
```suggestion
        if snapshot_ref := base_metadata.refs.get(self.ref):
```